cinema/permissions.py

```diff
@@ -0,0 +1,27 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    def has_permission(self, request, view):
+        return bool(
+            (
+                request.method in SAFE_METHODS
+                and request.user
+                and request.user.is_authenticated
+            )
+            or (request.user and request.user.is_staff)
+        )
+
+
+class IfAuthenticatedReadAndCreate(BasePermission):
+    def has_permission(self, request, view):
+        return bool(
+            (
+                (
+                    request.method in SAFE_METHODS or request.method == "POST"
+                )
+                and request.user
+                and request.user.is_authenticated
+            )
+            or (request.user and request.user.is_staff)
+        )
```

Comment:
No need to implement this permission class. Use default IsAuthenticated from rest_framework.permissions